pynucastro/rates/rate.py

```diff
@@ -630,40 +630,34 @@ def ydot_string_py(self):
         in a reaction network corresponding to this rate.
         """
 
-        # composition dependence
-        Y_string = ""
-        for n, r in enumerate(sorted(set(self.reactants))):
-            c = self.reactants.count(r)
-            if c > 1:
-                Y_string += f"Y[j{r}]**{c}"
-            else:
-                Y_string += f"Y[j{r}]"
+        ydot_string_component = []
 
-            if n < len(set(self.reactants))-1:
-                Y_string += "*"
+        # prefactor
+        if self.prefactor != 1.0:
+            ydot_string_component.append(f"{self.prefactor:1.14e}")
 
         # density dependence
-        if self.dens_exp == 0:
-            dens_string = ""
-        elif self.dens_exp == 1:
-            dens_string = "rho*"
-        else:
-            dens_string = f"rho**{self.dens_exp}*"
+        if self.dens_exp != 0:
```

Comment:
this condition can be moved to the else branch to reduce nesting (i.e. `elif self.dens_exp != 0:`)